frontend/src/metabase/static-viz/components/XYChart/utils/ticks.ts

```diff
@@ -91,7 +91,7 @@ export const getXTicksDimensions = (
     const rotatedSize = getRotatedXTickHeight(maxTextWidth);
 
     return {
-      width: Math.min(rotatedSize, MAX_ROTATED_TICK_WIDTH),
+      width: measureTextHeight(fontSize),
```

Comment:
This was my mistake, when I was changing the code from rotating x-axis tick labels 45 degrees to 90 degrees I didn't take this into account. Basically, when it's 45 degrees rotated, the width and height should be the same. But not when rotated 90 degrees, the x-axis tick label's original width become the height and its original height become the width.

#### After the fix
![image](https://user-images.githubusercontent.com/1937582/191992844-1d2e1b68-571b-4b17-8084-bf9391b5776b.png)


#### Before the fix
![image](https://user-images.githubusercontent.com/1937582/191992788-61650452-8064-4d2e-9c3d-f0f24bd7aa4d.png)